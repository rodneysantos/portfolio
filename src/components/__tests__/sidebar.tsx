import { cleanup, render } from "@testing-library/react";
import React from "react";
import rendeder from "react-test-renderer";
import Sidebar from "../sidebar";

afterEach(cleanup);

describe("Sidebar component", () => {
  it("renders correctly", () => {
    // act
    const tree = rendeder.create(<Sidebar />).toJSON();

    // assert
    expect(tree).toMatchSnapshot();
  });

  it("displays correct brand name", () => {
    // arrange
    const brandName = "XENON PHOTOGRAPHY";

    // act
    const { getByText } = render(<Sidebar />);

    // assert
    expect(getByText(brandName)).toBeDefined();
  });
});
