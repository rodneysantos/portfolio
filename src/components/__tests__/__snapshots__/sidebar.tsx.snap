// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component renders correctly 1`] = `
<div>
  <div>
    <div>
      Xenon Photography
    </div>
    <nav>
      <div>
        <a
          href="/gallery/black-and-white"
        >
          B & W
        </a>
      </div>
      <div>
        <a
          href="/gallery/color"
        >
          Color
        </a>
      </div>
      <div>
        <a
          href="/gallery/low-key"
        >
          Low-Key
        </a>
      </div>
      <div>
        <a
          href="/gallery/film-simulation"
        >
          Film Simulation
        </a>
      </div>
    </nav>
  </div>
</div>
`;
